dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -306,8 +306,6 @@ Spdp::write_secure_updates()
 {
   if (shutdown_flag_.value()) { return; }
 
-  ACE_GUARD(ACE_Thread_Mutex, g, lock_);
```

Comment:
Was this causing issues, or did it just get moved elsewhere?

Reply:
The same guard exists like one or two function calls up from this one. It was a guaranteed deadlock.

Reply:
But most of the time it the code wasn't deadlocking before, so there must have been something unique to this code path that was causing the issue, right? Did we normally just never test this code path? Do other methods that call into this function that call this code still need the lock here? If so, this exposes them to race conditions. Perhaps the code "one or two function calls up" needs to unlock before getting here, or we need to reassess our entry points?

Reply:
Changing QoS of an existing participant (with Secure Discovery) wasn't used before.

Reply:
Gotcha. This was the answer I was looking for!